src/pages/api/chat.ts

```diff
@@ -101,20 +102,51 @@ const handler = async (req: Request): Promise<NextResponse> => {
 }
 
 const refactorMessagesForOpenAI = (messages: Message[]): Message[] => {
```

Comment:
propose naming this `convertIConversatoinToOpenAI` with a little more type annotation in input and output.